src/diffusers/models/lora.py

```diff
@@ -352,16 +352,26 @@ def _unfuse_lora(self):
         self.w_down = None
 
     def forward(self, hidden_states: torch.Tensor, scale: float = 1.0) -> torch.Tensor:
-        if self.lora_layer is None:
-            # make sure to the functional Conv2D function as otherwise torch.compile's graph will break
-            # see: https://github.com/huggingface/diffusers/pull/4315
-            return F.conv2d(
-                hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
+        if self.padding_mode != "zeros":
+            hidden_states_pad = F.pad(hidden_states, self._reversed_padding_repeated_twice, mode=self.padding_mode)
+            original_outputs = F.conv2d(
+                hidden_states_pad,
+                self.weight,
+                self.bias,
+                self.stride,
+                (0, 0),
+                self.dilation,
+                self.groups,
             )
         else:
+            # make sure to the functional Conv2D function as otherwise torch.compile's graph will break
+            # see: https://github.com/huggingface/diffusers/pull/4315
             original_outputs = F.conv2d(
                 hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
             )
+        if self.lora_layer is None:
+            return original_outputs
+        else:
```

Comment:
```suggestion
        if self.padding_mode != "zeros":
            hidden_states = F.pad(hidden_states, self._reversed_padding_repeated_twice, mode=self.padding_mode)
            padding = (0, 0)
        else:
            padding = self.padding
     
        original_outputs = F.conv2d(
            hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
        )
        
        if self.lora_layer is None:
            return original_outputs
        else:
```
Wouldn't that be easier?

Reply:
Or is the padding wrong is this case?

Reply:
According to your opinion, the code on lines 361-363 should use padding instead of self.padding?

Reply:
Can you apply @patrickvonplaten 's suggestion here?
It does exactly the same thing but simplifies your code - that's all.

Reply:
Hi,I think his suggestion is great and agree with his modifications, but I think should use
```
original_outputs = F.conv2d(
             hidden_states, self.weight, self.bias, self.stride, padding, self.dilation, self.groups
         ) 
```
instead of
```
original_outputs = F.conv2d(
            hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
        )
```
Is that so?

Reply:
Gentle ping here: @yiyixuxu

Reply:
hey @jinghuan-Chen 
yes you're right!